java/simulated_ble/README.md

```diff
@@ -0,0 +1,306 @@
+
+```xml
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <!-- Inherit from parent -->
+  <parent>
+    <groupId>com.microsoft.azure.gateway</groupId>
+    <artifactId>gateway-module-base</artifactId>
+    <version>1.0.1</version>
+  </parent>
+  
+  <groupId>com.microsoft.azure.gateway</groupId>
+  <artifactId>ble-converter</artifactId>
+  <version>1.0</version>
+  <packaging>jar</packaging>
+
+  <!-- Set the filename of the Azure IoT Edge configuration located
+       under ./src/main/resources/gateway/ which is used in parent -->
+  <properties>
+    <gw.config.fileName>gw-config.json</gw.config.fileName>
+  </properties>
+
+  <!-- Re-declare dependencies used in parent -->
+  <dependencies>
+    <dependency>
+      <groupId>com.microsoft.azure.gateway</groupId>
+      <artifactId>gateway-java-binding</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>${dependency.runtime.group}</groupId>
+      <artifactId>${dependency.runtime.name}</artifactId>
+    </dependency>
+  </dependencies>
+
+  <!-- Re-declare plugins used in parent -->
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-resources-plugin</artifactId>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-shade-plugin</artifactId>
+      </plugin>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>exec-maven-plugin</artifactId>
+      </plugin>
+    </plugins>
+  </build>
+</project>
+```
+
+### Basic Knowledge of Module
+
+You could simply treat a module of Azure IoT Edge as a data processor whose job is to: receive input, process it, and produce output.
+
+The input might be data from hardware (like a motion detector), a message from other modules, or anything else (like a random number generated periodically by a timer).
+
+The output is similar to the input, it could trigger hardware behavior (like the blinking LED), a message to other modules, or anything else (like printing to the console).
+
+Modules communicate with each other using `com.microsoft.azure.gateway.messaging.Message` class. The **Content** of a `Message` is a byte array which is capable of representing any kind of data you like. **Properties** is also available in `Message`, it is simply a string-to-string map. You may think of Properties as the headers in HTTP request, or the metadata of a file.
```

Comment:
**Properties** are also available in the `Message` and are simply a string-to-string mapping. You may think of Properties as the headers in a HTTP request, or the metadata of a file.